### PR TITLE
python3Packages.quadprog: init at 0.1.11

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Quadratic Programming Solver.

https://github.com/quadprog/quadprog

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/quadprog/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, numpy
+}:
+
+buildPythonPackage rec {
+  pname = "quadprog";
+  version = "0.1.11";
```
Comment (fabaff) [inline pkgs/development/python-modules/quadprog/default.nix:9]: We should limit the supported Python version. The chance is very high that nobody will test the module on unsupported releases.

```suggestion
  version = "0.1.11";
  format = "setuptools";

  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/quadprog/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-jjIPn38UZhQeMX5hc26BCHIOatuuU/nWlg9ybv73iNk=";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
```
Comment (fabaff) [inline pkgs/development/python-modules/quadprog/default.nix:19]: The upstream repo contains [tests](https://github.com/quadprog/quadprog/tree/master/tests). If possible, the tests should be executed.

Please add `pythonImportsCheck`.

File: pkgs/development/python-modules/quadprog/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  meta = with lib; {
+    homepage = "https://github.com/quadprog/quadprog";
+    description = "Quadratic Programming Solver";
+    license = licenses.gpl2Only;
```
Comment (fabaff) [inline pkgs/development/python-modules/quadprog/default.nix:23]:
```suggestion
    license = licenses.gpl2Plus;
```

see https://github.com/quadprog/quadprog/blob/master/setup.py#L21

File: pkgs/development/python-modules/quadprog/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    inherit pname version;
+    hash = "sha256-jjIPn38UZhQeMX5hc26BCHIOatuuU/nWlg9ybv73iNk=";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/quadprog/quadprog";
```
Comment (fabaff) [inline pkgs/development/python-modules/quadprog/default.nix:21]:
```suggestion
    homepage = "https://github.com/quadprog/quadprog";
    changelog = "https://github.com/quadprog/quadprog/releases/tag/v${version}";
```

File: pkgs/development/python-modules/quadprog/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/python-modules/quadprog/default.nix:13]: Done.

File: pkgs/development/python-modules/quadprog/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/development/python-modules/quadprog/default.nix:32]: The tests don't seem to be present on PyPI. Switched to fetchFromGitHub for now.

**Comment by wegank** [comment]
@ofborg build python3Packages.quadprog

**Pull request merged**